src/services/topics.service.ts

```diff
@@ -0,0 +1,12 @@
+import { Service } from 'typedi';
+import { getRepository } from 'typeorm';
+import { Topic } from '@entities/topic.entity';
+
+@Service()
+export class TopicsService {
+  private readonly topicRepository = getRepository<Topic>(Topic);
+
+  listTopics(): Promise<Topic[]> {
```

Comment:
You could declare this function async in order to send it to the micro-task queue by force and not delegate this to the owner in using a promise-handling approach, what do you think?